experiment/gerrit-onboarder/onboarder.go

```diff
@@ -341,73 +349,131 @@ func ensureProjectConfig(config, host, cur_branch, groupName string) (string, er
+		return fmt.Errorf("failed to read project.config file: %w", err)
 	}
 
-	newData, err := ensureProjectConfig(string(data), host, cur_branch, groupName)
+	newData, err := ensureProjectConfig(workDir, string(data), host, cur_branch, groupName)
 	if err != nil {
 		return fmt.Errorf("failed to ensure updated project config: %w", err)
 	}
-	err = ioutil.WriteFile(projectConfigFile, []byte(newData), 0755)
+	err = ioutil.WriteFile(path.Join(workDir, projectConfigFile), []byte(newData), 0755)
 	if err != nil {
-		return fmt.Errorf("faield to write groups file: %w", err)
+		return fmt.Errorf("failed to write groups file: %w", err)
 	}
 
 	return nil
 }
 
-func fetchMetaConfig(host, repo, branch string) error {
-	if err := bumper.Call(os.Stdout, os.Stderr, "git", "fetch", fmt.Sprintf("sso://%s/%s", host, repo), "refs/meta/config"); err != nil {
+func fetchMetaConfig(host, repo, branch, workDir string) error {
+	if err := execInDir(os.Stdout, os.Stderr, workDir, "git", "fetch", fmt.Sprintf("sso://%s/%s", host, repo), "refs/meta/config"); err != nil {
 		return fmt.Errorf("failed to fetch refs/meta/config, %w", err)
 	}
-	if err := bumper.Call(os.Stdout, os.Stderr, "git", "checkout", "FETCH_HEAD"); err != nil {
+	if err := execInDir(os.Stdout, os.Stderr, workDir, "git", "checkout", "FETCH_HEAD"); err != nil {
 		return fmt.Errorf("failed to checkout FETCH_HEAD, %w", err)
 	}
-	if err := bumper.Call(os.Stdout, os.Stderr, "git", "switch", "-c", branch); err != nil {
+	if err := execInDir(os.Stdout, os.Stderr, workDir, "git", "switch", "-c", branch); err != nil {
 		return fmt.Errorf("failed to switch to new branch, %w", err)
 	}
 
 	return nil
 }
 
+func execInDir(stdout, stderr io.Writer, dir string, cmd string, args ...string) error {
+	(&logrus.Logger{
+		Out:       os.Stderr,
+		Formatter: logrus.StandardLogger().Formatter,
+		Hooks:     logrus.StandardLogger().Hooks,
+		Level:     logrus.StandardLogger().Level,
+	}).WithField("dir", dir).
+		WithField("cmd", cmd).
+		// The default formatting uses a space as separator, which is hard to read if an arg contains a space
+		WithField("args", fmt.Sprintf("['%s']", strings.Join(args, "', '"))).
+		Info("running command")
+
+	c := exec.Command(cmd, args...)
+	c.Dir = dir
+	c.Stdout = stdout
+	c.Stderr = stderr
+	return c.Run()
+}
+
+func createCR(workDir string) error {
+	diff, err := getDiff(workDir)
+	if err != nil {
+		return err
+	}
+	if diff == "" {
+		logrus.Info("No changes made. Returning without creating CR")
+		return nil
+	}
+	commitMessage := fmt.Sprintf("Grant the Prow cluster read and label permissions\n\nChange-Id: I%s", bumper.GitHash(fmt.Sprintf("%d", rand.Int())))
+	if err := execInDir(os.Stdout, os.Stderr, workDir, "git", "commit", "-a", "-v", "-m", commitMessage); err != nil {
+		return fmt.Errorf("unable to commit: %w", err)
+	}
+	if err := execInDir(os.Stdout, os.Stderr, workDir, "git", "push", "origin", "HEAD:refs/for/refs/meta/config"); err != nil {
```

Comment:
Would recommend moving dryrun check here for getting most out of dryrun

Reply:
Done